src/test/java/org/jitsi/utils/logging2/LoggerImplTest.java

```diff
@@ -186,6 +186,40 @@ public void testMsgSupplier()
         logger.info(() -> "hello, world!");
         assertEquals(1, fakeLogger.logLines.size());
     }
+
+    @Test
+    public void testHandler()
+    {
+        // We want to use a real logger for this test
+        LoggerImpl.loggerFactory = oldLoggerFactoryFunction;
+
+        LoggerImpl logger = new LoggerImpl("test");
+        FakeHandler handler = new FakeHandler();
+        logger.useHandler(handler);
+
+        logger.info("hello, world!");
+
+        assertEquals(1, handler.logRecords.size());
+        LogRecord record = handler.logRecords.get(0);
+        assertEquals(record.getMessage(), "hello, world!");
+        assertEquals(record.getLoggerName(), "test");
+    }
```

Comment:
Is there an easy way to test that the old parent handler didn't get used?

Reply:
Hm, not that I know of.  But since `setUseParentHandlers` is a direct pass through to the java.util.Logging method there's not too much room for things to go wrong.